src/coreclr/vm/frozenobjectheap.cpp

```diff
@@ -85,6 +85,15 @@ Object* FrozenObjectHeapManager::TryAllocateObject(PTR_MethodTable type, size_t
 #endif // !FEATURE_BASICFREEZE
 }
 
+// Does object belong to a frozen segment?
+bool FrozenObjectHeapManager::IsFromFrozenSegment(Object* object)
+{
+#ifdef FEATURE_BASICFREEZE
+    return GCHeapUtilities::GetGCHeap()->IsInFrozenSegment(object);
```

Comment:
Alternatively I can enumerate segments in FrozenObjectHeapManager and check if the pointer belongs to any

Reply:
This method won't be necessary if you incorporate my other feedback.